### PR TITLE
Fine Suara permissions control

### PR DESCRIPTION
**Pull request by tramuntanal**
#### :tophat: What? Why?
A basic improvement of security to avoid direct access to restricted contents as defined by Suara permissions.

Contents were already filtered and hidden to the users but this PR avoids direct access by hacking the url.

### CODE REVIEW

File: spec/controllers/components_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      end
+
+      let!(:space_with_other_permissions) do
+        create(
+          :consultation,
+          :published,
+          organization: organization,
+          suara_permissions: { ceco: "a", ceco_txt: "b" }
+        )
+      end
```
Comment (LauraJaimeC) [inline spec/controllers/components_controller_spec.rb:39]: Maybe you can test it too when space has one permission but not another, for example: 

```
let!(:space_with_one_permission) do
        create(
          :consultation,
          :published,
          organization: organization,
          suara_permissions: { ceco: "ceco", ceco_txt: "b" }
        )
      end
```

File: spec/controllers/comments_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      end
+
+      let!(:space_without_permissions) do
+        create(
+          :participatory_process,
+          :published,
+          organization: organization,
+          suara_permissions: { ceco: "", ceco_txt: "", tipologia: "" }
+        )
+      end
```
Comment (LauraJaimeC) [inline spec/controllers/comments_controller_spec.rb:38]: Maybe you can test it too when space has one permission but not another, for example: 

```
let!(:space_with_one_permission) do
        create(
          :consultation,
          :published,
          organization: organization,
          suara_permissions: { ceco: "ceco", ceco_txt: "b" }
        )
      end
```

File: app/decorators/controllers/decidim/assemblies/assemblies_controller_decorator.rb [hunk 3]
```diff
@@ -3,7 +3,13 @@
 # This decorator change the behavior of the promoted and parent assemblies
 # to check permissions if you are not an admin user.
 Decidim::Assemblies::AssembliesController.class_eval do
-  include FilterParticipatorySpacesHelper
+  include SuaraPermissionsSupervisor
+
+  alias_method :original_show, :show
+  def show
+    original_show
+    render status: :forbidden unless current_user.admin? || suara_permissions_match?(current_user, current_participatory_space)
```
Comment (LauraJaimeC) [inline app/decorators/controllers/decidim/assemblies/assemblies_controller_decorator.rb:11]: It is a good idea to extract that condition in an method? What do you think?

**Review by LauraJaimeC — CHANGES_REQUESTED** [review]
Good job!! 💪🏼  Just a few details!

**Comment by tramuntanal** [comment]
Changes applied @LauraJaimeC

**Review by LauraJaimeC — APPROVED** [review]
LGTM!!

**Pull request merged**